subworkflows/local/analyse_transcripts.nf

```diff
@@ -56,9 +57,19 @@ workflow ANALYSE_TRANSCRIPTS {
             drop_group_samples_as
         )
 
-        // Generates  config file and runs Aberrant expression module
-        ch_bai_files = ch_bam_ds_bai.collect{ it[2] }.toList()
-        ch_bam_bai_files = ch_bam_files.toList().combine(ch_bai_files)
+
+        // Generates config file and runs Aberrant expression module
+
+        // Sort bam and bai files for stable order
+        ch_bam_files_sorted = ch_bam_files.toList().map { list ->
+            list.sort { a, b -> a.getName() <=> b.getName() }
+        }
+        ch_bai_files_sorted = ch_bam_ds_bai.collect{ it[2] }.toList().map { list ->
+            list.sort { a, b -> a.getName() <=> b.getName() }
+        }
+        ch_bam_bai_files = ch_bam_files_sorted.combine(ch_bai_files_sorted)
```

Comment:
```suggestion
        ch_bam_files_annot
            .map { id, single_end, strandedness, sex, bam, bai ->
            [ bam, bai ] 
            }
        .set{ ch_bam_bai_files }
```

Reply:
What if we would do something like here to make it a bit more concise and to make sure that both channels are always in the same order?

Reply:
Yes, that looks much more elegant. I'll give it a run and see if it works.